packages/medusa/src/commands/develop.js

```diff
@@ -23,6 +23,109 @@ export default async function ({ port, directory }) {
   args.shift()
   args.shift()
 
+  /**
+   * Re-constructing the path to Medusa CLI to execute the
+   * start command.
+   */
+  const cliPath = path.resolve(
+    require.resolve("@medusajs/medusa-cli"),
+    "..",
+    "..",
+    "cli.js"
+  )
+
+  const devServer = {
+    childProcess: null,
+    watcher: null,
+
+    /**
+     * Start the development server by forking a new process.
+     *
+     * We do not kill the parent process when child process dies. This is
+     * because sometimes the dev server can die because of programming
+     * or logical errors and we can still watch the file system and
+     * restart the dev server instead of asking the user to re-run
+     * the command.
+     */
+    start() {
+      this.childProcess = fork(cliPath, ["start", ...args], {
+        cwd: directory,
+        env: {
+          ...process.env,
+          NODE_ENV: "development",
+        },
+        execArgv: argv,
+      })
+      this.childProcess.on("error", (error) => {
+        Logger.error("Dev server died with error", error)
+        Logger.info("Still watching filesystem to restart the dev server")
+      })
+    },
+
+    /**
+     * Restarts the development server by cleaning up the existing
+     * child process and forking a new one
+     */
+    restart() {
+      if (this.childProcess) {
+        this.childProcess.removeAllListeners()
+        if (process.platform === "win32") {
+          execSync(`taskkill /PID ${this.childProcess.pid} /F /T`)
+        }
+        this.childProcess.kill("SIGINT")
+      }
+      this.start()
+    },
+
+    /**
+     * Watches the entire file system and ignores the following files
+     *
+     * - Dot files
+     * - node_modules
+     * - dist
+     * - src/admin/**
+     */
+    watch() {
+      this.watcher = chokidar.watch(["."], {
+        ignoreInitial: true,
+        cwd: process.cwd(),
+        ignored: [/(^|[\\/\\])\../, "node_modules", "dist", "src/admin/**/*"],
+      })
+
+      this.watcher.on("add", (file) => {
+        Logger.info(
+          `${path.relative(
+            directory,
+            file
+          )} file created: restarting dev server...`
```

Comment:
Suggestion: we can maybe remove the file word? So that it looks more like 
service.ts created